store/store_test.go

```diff
@@ -907,3 +893,378 @@ func TestStore_HasAt(t *testing.T) {
 	has = store.HasAt(ctx, 0)
 	assert.False(t, has)
 }
+
+func TestStore_DeleteRange(t *testing.T) {
```

Comment:
can we please add test cases for partial deletes here (in case of caller ctx timeout or errors) that the store recovers properly?